presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -597,7 +654,6 @@ public MethodHandleAndConstructor specialize(Signature boundSignature, BoundVari
             if (nullable) {
                 return Primitives.wrap(clazz);
             }
-            checkArgument(clazz != void.class);
```

Comment:
Looking at the code it seems like we should disallow both of them. Having a `void` native container type combined with a non-nullable parameter/return value doesn't make sense, as `NULL` is that only value that can be represented in a `void`